journal-writer/src/main/scala/akka/stream/alpakka/persistence/writer/JournalWriter.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.persistence.writer
+
+import akka.actor.{ Actor, ActorRef, ActorSystem, Props }
+import akka.pattern.ask
+import akka.persistence.Persistence
+import akka.persistence.journal.writer.WriteJournalAdapterCameo
+import akka.persistence.query.{ EventEnvelope, EventEnvelope2 }
+import akka.stream.scaladsl._
+import akka.util.Timeout
+import akka.{ Done, NotUsed }
+
+import scala.collection.immutable._
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.reflect.ClassTag
+
+class WriteJournalAdapter(writePlugin: ActorRef) extends Actor {
+  def cameo(replyTo: ActorRef, writePlugin: ActorRef, messages: Seq[EventEnvelope]): ActorRef =
```

Comment:
As far as I can see it's stateless and doesn't do anything that can't be done in the adapter actor, but perhaps the mapping of the replies is the motivation for it.